src/Orleans/Streams/Internal/StreamConsumerExtension.cs

```diff
@@ -67,13 +67,18 @@ internal bool RemoveObserver(GuidId subscriptionId)
             return allStreamObservers.TryRemove(subscriptionId, out ignore);
         }
 
-        public Task<StreamHandshakeToken> DeliverItem(GuidId subscriptionId, Immutable<object> item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
+        public Task<StreamHandshakeToken> DeliverImmutable(GuidId subscriptionId, Immutable<object> item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
         {
-            if (logger.IsVerbose3) logger.Verbose3("DeliverItem {0} for subscription {1}", item.Value, subscriptionId);
+            return DeliverMutable(subscriptionId, item.Value, currentToken, handshakeToken);
+        }
+
+        public Task<StreamHandshakeToken> DeliverMutable(GuidId subscriptionId, object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
+        {
+            if (logger.IsVerbose3) logger.Verbose3("DeliverItem {0} for subscription {1}", item, subscriptionId);
```

Comment:
I understand that you're just preserving the existing logging, but not knowing the type of item, i'd be very hesitant to log the actual item.